File: app/src/main/java/com/booboot/vndbandroid/bean/vndb/VNDBCommand.java
package com.booboot.vndbandroid.bean.vndb;

import java.io.Serializable;
import java.util.HashMap;

/**
 * Created by od on 12/03/2016.
 */
public abstract class VNDBCommand implements Serializable {
    private final static HashMap<String, Class> CLASSES = new HashMap<>();

    static {
        CLASSES.put("login", Login.class);
        CLASSES.put("error", Error.class);
        CLASSES.put("results", Results.class);
        CLASSES.put("dbstats", DbStats.class);
    }

    public static Class getClass(String command) {
        return CLASSES.get(command);
    }
}

File: app/src/main/java/com/booboot/vndbandroid/model/vndb/VNDBCommand.kt
package com.booboot.vndbandroid.model.vndb

import java.io.Serializable

abstract class VNDBCommand : Serializable {
    companion object {
        private val CLASSES = mapOf(
                "login" to Login::class.java,
                "error" to Error::class.java,
                "results" to Results::class.java,
                "dbstats" to DbStats::class.java
        )

        fun getClass(command: String) = CLASSES[command]
    }
}
